### PR TITLE
feat(cli): accept stdin

### LINKED ISSUE #757: Accept input via stdin
Accept stdin as an alternative way to get a document, so that folks can use spectral in the middle of complex workflows without needing to write temporary files. 

```bash
cat openapi.yml | spectral -r some-ruleset.yml
```

This was mention by @radicarl in https://github.com/stoplightio/spectral/issues/656#issuecomment-550960358 but has been mentioned a few times in person and elsewhere. Just kept forgetting to write it down.

### PR DESCRIPTION
**Pull request by P0lip**
Closes #757

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

~Added isTTY env variable.
https://github.com/stoplightio/spectral/tree/chore/test-harness work has been pushed here. Still need to handle tty correctly, but it's getting closed.~

### CODE REVIEW

File: src/cli/services/linter/utils/listFiles.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import { normalize } from '@stoplight/path';
 import * as fg from 'fast-glob';
 
-export async function listFiles(pattens: string[]): Promise<string[]> {
-  const { files, urls } = pattens.reduce<{ files: string[]; urls: string[] }>(
+export async function listFiles(pattens: Array<number | string>): Promise<Array<number | string>> {
+  const { files, fds, urls } = pattens.reduce<{ files: string[]; urls: string[]; fds: number[] }>(
```
Comment (philsturgeon) [inline src/cli/services/linter/utils/listFiles.ts]: Could we rename fds to fileDescriptors or something? I'm not sure what a file descriptor is but that's more readable than fds.

File: src/cli/commands/lint.ts [hunk 2]
```diff
@@ -14,13 +15,25 @@ const formatOptions = Object.values(OutputFormat);
...
       .positional('documents', {
         description:
           'Location of JSON/YAML documents. Can be either a file, a glob or fetchable resource(s) on the web.',
         type: 'string',
+        coerce(values) {
+          if (values.length > 0) {
+            return values;
+          }
+
+          if (process.env.isTTY || process.stdin.isTTY) {
```
Comment (philsturgeon) [inline src/cli/commands/lint.ts]: Im not sure I understand what is happening here? What is TTY and why does it need to be true?

File: test-harness/scenarios/help-no-document.scenario [hunk 3]
```diff
@@ -2,13 +2,15 @@
 Errors when no document is provided
 ====command====
 lint
+====env====
+isTTY=true
```
Comment (philsturgeon) [inline test-harness/scenarios/help-no-document.scenario]: Could we get a test harness test showing `cat somefile.yaml | spectral --something` to show that its working as expected?

File: src/cli/commands/lint.ts (hunk 2 above)
Comment (P0lip) [reply src/cli/commands/lint.ts]: I added comments explaining that. LMK if it helps.

**Pull request merged**